Resources/doc/task-crud-tutorial.rst

```diff
@@ -0,0 +1,336 @@
+        {
+            $this->name = $name;
+
+            return $this;
+        }
+
+        /**
+         * Get name
+         *
+         * @return string
+         */
+        public function getName()
+        {
+            return $this->name;
+        }
+
+        /**
+         * Set code
+         *
+         * @param string $completed
+         *
+         * @return Task
+         */
+        public function setCompleted($completed)
+        {
+            $this->completed = $completed;
+
+            return $this;
+        }
+
+        /**
+         * Get code
+         *
+         * @return string
+         */
+        public function isCompleted()
+        {
+            return $this->completed;
+        }
+    }
+
+B) Create TaskType form
+-----------------------
+
+For handling data comes from request we need to create our ``TaskType.php`` file with standard content
+(pleas note that we setup ``getName()`` method with ``'task'`` and ``'csrf_protection => false'`` - this is
+important for creating requests which we will do later):
+
+.. code-block:: php
+
+    // src/AppBundle/Form/TaskType.php
+    namespace AppBundle\Form;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    class TaskType extends AbstractType
+    {
+        /**
+         * @param FormBuilderInterface $builder
+         * @param array $options
+         */
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('name')
+                ->add('completed')
+            ;
+        }
+
+        /**
+         * @param OptionsResolver $resolver
+         */
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'AppBundle\Entity\Task',
+                'csrf_protection' => false,
+            ));
+        }
+
+        /**
+         * @return string
+         */
+        public function getName()
+        {
+            return 'task';
+        }
+    }
+
+C) Create TaskController
+------------------------
+
+For expose our REST API methods (routes) lets add the following controller:
+
+.. code-block:: php
+
+    // src/AppBundle/Controller/TaskController.php
+    namespace AppBundle\Controller;
+
+    use AppBundle\Entity\Task;
+    use AppBundle\Form\TaskType;
+    use FOS\RestBundle\Controller\Annotations\View;
+    use FOS\RestBundle\Controller\FOSRestController;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+    class TaskController extends FOSRestController
+    {
+        /**
+         * List all tasks
+         *
+         * @View
+         */
+        public function getTasksAction()
+        {
+            return $this->getDoctrine()->getRepository('AppBundle:Task')->findAll();
```

Comment:
I would add a `private` `getRepository()` method and use it throughout this controller to short the code.